src/librustc_mir/interpret/eval_context.rs

```diff
@@ -100,12 +116,15 @@ pub enum StackPopCleanup {
     /// Jump to the next block in the caller, or cause UB if None (that's a function
     /// that may never return). Also store layout of return place so
     /// we can validate it at that layout.
-    Goto(Option<mir::BasicBlock>),
+    /// 'ret' stores the block we jump to on a normal return, while 'unwind'
+    /// stores the block used for cleanup during unwinding
+    Goto { ret: Option<mir::BasicBlock>, unwind: Option<mir::BasicBlock> },
     /// Just do nohing: Used by Main and for the box_alloc hook in miri.
     /// `cleanup` says whether locals are deallocated. Static computation
     /// wants them leaked to intern what they need (and just throw away
     /// the entire `ecx` when it is done).
-    None { cleanup: bool },
+    /// `wundin` stores the block used for cleanup during unwinding
+    None { cleanup: bool, unwind: Option<mir::BasicBlock> },
```

Comment:
This very special `StackPopCleanup` is only used for a few cases as indicated, and I don't think there is a possibility of unwinding there. So why is an `unwind` field needed in `None`?

`exchange_malloc` even has a comment saying "This function must not unwind.", and `start_fn` fairly immediately calls `catch_unwind`.

Reply:
I can remove it.

Reply:
Please do.